dot/state/grandpa.go

```diff
@@ -257,9 +257,10 @@ func (s *GrandpaState) ApplyForcedChanges(importedBlockHeader *types.Header) err
 		return fmt.Errorf("cannot set change set id at block")
 	}
 
-	logger.Debugf("Applying authority set forced change at block #%d",
-		forcedChange.announcingHeader.Number)
+	logger.Debugf("Applied authority set forced change: %s", forcedChange)
 
+	s.forcedChanges.pruneAll()
```

Comment:
Can there be conflicts when applying a forced change and pruning other forced changes? Does it only prune changes that were requested before the current change?

Reply:
> Does it only prune changes that were requested before the current change

Reply:
>Does it only prune changes that were requested before the current change?

No, it prunes the entire set, substrate does it as well https://github.com/paritytech/substrate/blob/50de15d8740a129db9c18a6698fbd183b00326a2/client/consensus/grandpa/src/authorities.rs#L511. 
A forced change once applied should change the entire authority set for all forks https://spec.polkadot.network/#sect-finality-forced-changes

Reply:
awesome thank you